### DISCUSSION TITLE
when i exec master.shutdown() the port not be free

### DISCUSSION THREAD

**Discussion by DspringL** [mitmproxy/mitmproxy#6830] [General]
async def start_proxy(host: str = '0.0.0.0', port: int = 9090) -> Master:
       opts = options.Options(listen_host=host, listen_port=port)
       master = dump.DumpMaster(
            opts,
            with_termlog=True,
            with_dumper=True,
       )
       master.addons.add(HttpHook())
       await master.run()
       return master

    proxy_loop = asyncio.new_event_loop()
    coro = start_proxy()
    proxy_loop.run_until_complete(coro)
    ctx.master.shutdown()

<img width="560" alt="image" src="https://github.com/mitmproxy/mitmproxy/assets/29829184/9933bbfc-eb02-4f5c-9f25-bd69818e14b4">

#### System Information
Mitmproxy: 10.3.0
Python:    3.11.9
OpenSSL:   OpenSSL 3.2.1 30 Jan 2024
Platform:  macOS-13.5.2-arm64-arm-64bit

**Comment by DspringL** [comment]
The shutdown() port has not been released, expected the port to be released

**Reply by meilulin** [reply]
How to solve this? or Close it using other methods?
